backupccl: `TestFullClusterBackup/ensure_that_jobs_are_restored` flakes under span configs

**Describe the problem**

Saw a `TestFullClusterBackup/ensure_that_jobs_are_restored` failure [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=4142848&buildTypeId=Cockroach_UnitTests) that looks like fallout from https://github.com/cockroachdb/cockroach/pull/73876. This issue tracks the investigation for as to why.

```
------- Stdout: -------
=== RUN   TestFullClusterBackup/ensure_that_jobs_are_restored
    full_cluster_backup_restore_test.go:334: 
        	Error Trace:	full_cluster_backup_restore_test.go:334
        	Error:      	Not equal: 
        	            	expected: []string{"728894745672974337", "running", "2022-01-18 13:10:23.459824 +0000 +0000", "\n\x1freconciling span configurations\x12\x04root\x18\x9b\xc6\xcaﯻ\xf5\x02\xa0\x01\x01\xda\x01\x00", "\x10\xf5\xb3\xc6ﯻ\xf5\x02\xa8\x01\xf2\xf4\xc1\xe9\xb0ւ\xfes\xb2\x01\x02\n\x00", "NULL", "NULL", "1"}
        	            	actual  : []string{"728894745672974337", "running", "2022-01-18 13:10:23.459824 +0000 +0000", "\n\x1freconciling span configurations\x12\x04root\x18\x9b\xc6\xcaﯻ\xf5\x02\xa0\x01\x01\x82\x02\x89!\n\arunning\x10鳥\xfe\xaf\xbb\xf5\x02\x18֝\xe8\xfe\xaf\xbb\xf5\x02 \x01*\xe9 \x12\xe6 \n\xf9\x1a\x12\xf6\x1a\n\xe4\x18\x12\xe1\x18\n\xd3\x17\x12\xd0\x17\n\xb4\x02\n\xb1\x02\n(expected to delete 11 row(s), deleted 10\x12\x84\x02\n8github.com/cockroachdb/errors/errutil/*errutil.leafError\x12:\n8github.com/cockroachdb/errors/errutil/*errutil.leafError\x1a(expected to delete 11 row(s), deleted 10\"b\n4type.googleapis.com/cockroach.errorspb.StringPayload\x12*\n(expected to delete 11 row(s), deleted 10\x1a\x96\x15\n<github.com/cockroachdb/errors/withstack/*withstack.withStack\x12>\n<github.com/cockroachdb/errors/withstack/*withstack.withStack\x1a\x95\x14\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor.(*KVAccessor).updateSpanConfigEntriesWithTxn\n\t/go/src/github.com/cockroachd
b/cockroach/pkg/spanconfig/spanconfigkvaccessor/kvaccessor.go:186\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor.(*KVAccessor).UpdateSpanConfigEntries.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor/kvaccessor.go:128\ngithub.com/cockroachdb/cockroach/pkg/kv.runTxn.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849\ngithub.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:972\ngithub.com/cockroachdb/cockroach/pkg/kv.runTxn\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848\ngithub.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor.(*KVAccessor).UpdateSpanConfigEntries\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor/kvaccessor.go:127\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*fullReconciler).d
eleteExtraneousSpanConfigs\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler/reconciler.go:330\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*fullReconciler).reconcile\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler/reconciler.go:253\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*Reconciler).Reconcile\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler/reconciler.go:150\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigjob.(*resumer).Resume\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigjob/job.go:74\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1123\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1124\ngithub.com/cockroachdb/cockroach/pkg/jobs.
(*Registry).runJob\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:401\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:323\ngithub.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:488\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581\x1a\x88\x01\nAgithub.com/cockroachdb/errors/assert/*assert.withAssertionFailure\x12C\nAgithub.com/cockroachdb/errors/assert/*assert.withAssertionFailure\x128non-cancelable: expected to delete 11 row(s), deleted 10\x1a\xd2\x01\n9github.com/cockroachdb/errors/errutil/*errutil.withPrefix\x12;\n9github.com/cockroachdb/errors/errutil/*errutil.withPrefix\x1a\x0enon-cancelable\"H\n4type.googleapis.com/cockroach.errorspb.StringPayload\x12\x10\n\x0enon-cancelable\x1a\xe7\x05\n<github.com/cockroachdb/errors/withstack/*withstack.withStack\x12>\n<github.com/cockroachdb/errors/withstack/*withs
tack.withStack\x1a\xe6\x04\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1148\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:401\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:323\ngithub.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:488\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581\xda\x01\x00", "\x10\x84\xfb\xa0\xfe\xaf\xbb\xf5\x02\xa8\x01\xec\xda\xea\xb3ó\xea\x9dv\xb2\x01\x02\n\x00", "NULL", "NULL", "1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -4,4 +4,4 @@
        	            	  (string) (len=38) "2022-01-18 13:10:23.459824 +0000 +0000",
        	            	- (string) (len=54) "\n\x1freconciling span configurations\x12\x04root\x18\x9b\xc6\xcaﯻ\xf5\x02\xa0\x01\x01\xda\x01\x00",
        	            	- (string) (len=25) "\x10\xf5\xb3\xc6ﯻ\xf5\x02\xa8\x01\xf2\xf4\xc1\xe9\xb0ւ\xfes\xb2\x01\x02\n\x00",
        	            	+ (string) (len=4291) "\n\x1freconciling span configurations\x12\x04root\x18\x9b\xc6\xcaﯻ\xf5\x02\xa0\x01\x01\x82\x02\x89!\n\arunning\x10鳥\xfe\xaf\xbb\xf5\x02\x18֝\xe8\xfe\xaf\xbb\xf5\x02 \x01*\xe9 \x12\xe6 \n\xf9\x1a\x12\xf6\x1a\n\xe4\x18\x12\xe1\x18\n\xd3\x17\x12\xd0\x17\n\xb4\x02\n\xb1\x02\n(expected to delete 11 row(s), deleted 10\x12\x84\x02\n8github.com/cockroachdb/errors/errutil/*errutil.leafError\x12:\n8github.com/cockroachdb/errors/errutil/*errutil.leafError\x1a(expected to delete 11 row(s), deleted 10\"b\n4type.googleapis.com/cockroach.errorspb.StringPayload\x12*\n(expected to delete 11 row(s), deleted 10\x1a\x96\x15\n<github.com/cockroachdb/errors/withstack/*withstack.withStack\x12>\n<github.com/cockroachdb/errors/withstack/*withstack.withStack\x1a\x95\x14\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor.(*KVAccessor).updateSpanConfigEntriesWithTxn\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor/kvaccessor.go:186\ngithu
b.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor.(*KVAccessor).UpdateSpanConfigEntries.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor/kvaccessor.go:128\ngithub.com/cockroachdb/cockroach/pkg/kv.runTxn.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849\ngithub.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:972\ngithub.com/cockroachdb/cockroach/pkg/kv.runTxn\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848\ngithub.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn\n\t/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor.(*KVAccessor).UpdateSpanConfigEntries\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigkvaccessor/kvaccessor.go:127\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*fullReconciler).deleteExtraneousSpanConfigs\n\t/go/src/github.com/cockroachdb/cockroach/p
kg/spanconfig/spanconfigreconciler/reconciler.go:330\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*fullReconciler).reconcile\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler/reconciler.go:253\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler.(*Reconciler).Reconcile\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigreconciler/reconciler.go:150\ngithub.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigjob.(*resumer).Resume\n\t/go/src/github.com/cockroachdb/cockroach/pkg/spanconfig/spanconfigjob/job.go:74\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1123\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1124\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/
adopt.go:401\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:323\ngithub.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:488\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581\x1a\x88\x01\nAgithub.com/cockroachdb/errors/assert/*assert.withAssertionFailure\x12C\nAgithub.com/cockroachdb/errors/assert/*assert.withAssertionFailure\x128non-cancelable: expected to delete 11 row(s), deleted 10\x1a\xd2\x01\n9github.com/cockroachdb/errors/errutil/*errutil.withPrefix\x12;\n9github.com/cockroachdb/errors/errutil/*errutil.withPrefix\x1a\x0enon-cancelable\"H\n4type.googleapis.com/cockroach.errorspb.StringPayload\x12\x10\n\x0enon-cancelable\x1a\xe7\x05\n<github.com/cockroachdb/errors/withstack/*withstack.withStack\x12>\n<github.com/cockroachdb/errors/withstack/*withstack.withStack\x1a\xe6\x04\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*
Registry).stepThroughStateMachine\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1148\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:401\ngithub.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1\n\t/go/src/github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:323\ngithub.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2\n\t/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:488\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581\xda\x01\x00",
        	            	+ (string) (len=25) "\x10\x84\xfb\xa0\xfe\xaf\xbb\xf5\x02\xa8\x01\xec\xda\xea\xb3ó\xea\x9dv\xb2\x01\x02\n\x00",
        	            	  (string) (len=4) "NULL",
        	Test:       	TestFullClusterBackup/ensure_that_jobs_are_restored
    --- FAIL: TestFullClusterBackup/ensure_that_jobs_are_restored (0.03s)
```


**To Reproduce**

```
dev test pkg/ccl/backupccl -f=TestFullClusterBackup/ensure_that_jobs_are_restored
```

Jira issue: CRDB-12446

--- a/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
+++ b/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
@@ -50,16 +50,12 @@ func TestFullClusterBackup(t *testing.T) {
 
 	params := base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
-			DisableSpanConfigs: true, // TODO(irfansharif): #75060.
 			Knobs: base.TestingKnobs{
 				SpanConfig: &spanconfig.TestingKnobs{
 					// We compare job progress before and after a restore. Disable
 					// the automatic jobs checkpointing which could possibly mutate
 					// the progress data during the backup/restore process.
 					JobDisablePersistingCheckpoints: true,
-				},
-				GCJob: &sql.GCJobTestingKnobs{
-					DisableNewProtectedTimestampSubsystemCheck: true,
 				},
 			},
 		}}
@@ -91,10 +87,17 @@ func TestFullClusterBackup(t *testing.T) {
 	}
 
 	// The claim_session_id field in jobs is a uuid and so needs to be excluded
-	// when comparing jobs pre/post restore.
+	// when comparing jobs pre/post restore. The span config reconciliation job
+	// too is something we exclude; because it's a singleton job, when restored
+	// into another cluster it self-terminates.
 	const jobsQuery = `
 SELECT id, status, created, payload, progress, created_by_type, created_by_id, claim_instance_id
 FROM system.jobs
+WHERE id NOT IN
+(
+	SELECT job_id FROM [SHOW AUTOMATIC JOBS]
+  WHERE job_type = 'AUTO SPAN CONFIG RECONCILIATION'
+)
 	`
 	// Pause SQL Stats compaction job to ensure the test is deterministic.
 	sqlDB.Exec(t, `PAUSE SCHEDULES SELECT id FROM [SHOW SCHEDULES FOR SQL STATISTICS]`)

--- a/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
+++ b/pkg/ccl/backupccl/full_cluster_backup_restore_test.go
@@ -30,6 +30,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/systemschema"
 	"github.com/cockroachdb/cockroach/pkg/sql/execinfra"
 	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util"
@@ -368,6 +369,42 @@ CREATE TABLE data2.foo (a int);
 			fmt.Sprintf("SELECT id FROM system.namespace WHERE name = '%s'", tableName)).Scan(&tableID)
 		require.True(t, tableID > maxID)
 		require.NotEqual(t, dbID, tableID)
+	})
+}
+
+// TestSingletonSpanConfigJobPostRestore ensures that there's a single span
+// config reconciliation job running post restore.
+func TestSingletonSpanConfigJobPostRestore(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	params := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			},
+		},
+	}
+	const numAccounts = 10
+	_, sqlDB, tempDir, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, numAccounts, InitManualReplication, params)
+	_, sqlDBRestore, cleanupEmptyCluster := backupRestoreTestSetupEmpty(t, singleNode, tempDir, InitManualReplication, params)
+	defer cleanupFn()
+	defer cleanupEmptyCluster()
+
+	sqlDB.Exec(t, `BACKUP TO $1`, localFoo)
+	sqlDBRestore.Exec(t, `RESTORE FROM $1`, localFoo)
+
+	const numRunningReconciliationJobQuery = `
+SELECT count(*) FROM [SHOW AUTOMATIC JOBS]
+WHERE job_type = 'AUTO SPAN CONFIG RECONCILIATION' AND status = 'running'
+`
+	testutils.SucceedsSoon(t, func() error {
+		var numRunningJobs int
+		sqlDBRestore.QueryRow(t, numRunningReconciliationJobQuery).Scan(&numRunningJobs)
+		if numRunningJobs != 1 {
+			return errors.Newf("expected single running reconciliation job, found %d", numRunningJobs)
+		}
+		return nil
 	})
 }
 

--- a/pkg/jobs/jobspb/jobs.proto
+++ b/pkg/jobs/jobspb/jobs.proto
@@ -16,6 +16,7 @@ import "errorspb/errors.proto";
 import "gogoproto/gogo.proto";
 import "roachpb/api.proto";
 import "roachpb/data.proto";
+import "roachpb/metadata.proto";
 import "roachpb/io-formats.proto";
 import "sql/catalog/descpb/structured.proto";
 import "sql/catalog/descpb/tenant.proto";
@@ -961,7 +962,19 @@ message Payload {
   // the jobs.execution_errors.max_entries cluster setting.
   repeated RetriableExecutionFailure retriable_execution_failure_log = 32;
 
-  // NEXT ID: 34.
+  // CreationClusterID is populated at creation with the ClusterID, in case a
+  // job resuming later, needs to use this information, e.g. to determine if it
+  // has been restored into a different cluster, which might mean it should
+  // terminate, pause or update some other state.
+  bytes creation_cluster_id = 35 [(gogoproto.nullable) = false, (gogoproto.customname) = "CreationClusterID",
+    (gogoproto.customtype) = "github.com/cockroachdb/cockroach/pkg/util/uuid.UUID"];
+
+  // CreationClusterVersion is populated at creation time with the then-active
+  // cluster version, in case a job resuming later needs to use this information
+  // to migrate or update the job.
+  roachpb.Version creation_cluster_version = 36 [(gogoproto.nullable) = false];
+
+  // NEXT ID: 37.
 }
 
 message Progress {

--- a/pkg/jobs/registry.go
+++ b/pkg/jobs/registry.go
@@ -89,17 +89,18 @@ const (
 type Registry struct {
 	serverCtx context.Context
 
-	ac       log.AmbientContext
-	stopper  *stop.Stopper
-	db       *kv.DB
-	ex       sqlutil.InternalExecutor
-	clock    *hlc.Clock
-	nodeID   *base.SQLIDContainer
-	settings *cluster.Settings
-	execCtx  jobExecCtxMaker
-	metrics  Metrics
-	td       *tracedumper.TraceDumper
-	knobs    TestingKnobs
+	ac        log.AmbientContext
+	stopper   *stop.Stopper
+	db        *kv.DB
+	ex        sqlutil.InternalExecutor
+	clock     *hlc.Clock
+	clusterID *base.ClusterIDContainer
+	nodeID    *base.SQLIDContainer
+	settings  *cluster.Settings
+	execCtx   jobExecCtxMaker
+	metrics   Metrics
+	td        *tracedumper.TraceDumper
+	knobs     TestingKnobs
 
 	// adoptionChan is used to nudge the registry to resume claimed jobs and
 	// potentially attempt to claim jobs.
@@ -182,6 +183,7 @@ func MakeRegistry(
 	clock *hlc.Clock,
 	db *kv.DB,
 	ex sqlutil.InternalExecutor,
+	clusterID *base.ClusterIDContainer,
 	nodeID *base.SQLIDContainer,
 	sqlInstance sqlliveness.Instance,
 	settings *cluster.Settings,
@@ -198,6 +200,7 @@ func MakeRegistry(
 		clock:               clock,
 		db:                  db,
 		ex:                  ex,
+		clusterID:           clusterID,
 		nodeID:              nodeID,
 		sqlInstance:         sqlInstance,
 		settings:            settings,
@@ -275,27 +278,29 @@ func (r *Registry) MakeJobID() jobspb.JobID {
 }
 
 // newJob creates a new Job.
-func (r *Registry) newJob(record Record) *Job {
+func (r *Registry) newJob(ctx context.Context, record Record) *Job {
 	job := &Job{
 		id:        record.JobID,
 		registry:  r,
 		createdBy: record.CreatedBy,
 	}
-	job.mu.payload = r.makePayload(&record)
+	job.mu.payload = r.makePayload(ctx, &record)
 	job.mu.progress = r.makeProgress(&record)
 	job.mu.status = StatusRunning
 	return job
 }
 
 // makePayload creates a Payload structure based on the given Record.
-func (r *Registry) makePayload(record *Record) jobspb.Payload {
+func (r *Registry) makePayload(ctx context.Context, record *Record) jobspb.Payload {
 	return jobspb.Payload{
-		Description:   record.Description,
-		Statement:     record.Statements,
-		UsernameProto: record.Username.EncodeProto(),
-		DescriptorIDs: record.DescriptorIDs,
-		Details:       jobspb.WrapPayloadDetails(record.Details),
-		Noncancelable: record.NonCancelable,
+		Description:            record.Description,
+		Statement:              record.Statements,
+		UsernameProto:          record.Username.EncodeProto(),
+		DescriptorIDs:          record.DescriptorIDs,
+		Details:                jobspb.WrapPayloadDetails(record.Details),
+		Noncancelable:          record.NonCancelable,
+		CreationClusterVersion: r.settings.Version.ActiveVersion(ctx).Version,
+		CreationClusterID:      r.clusterID.Get(),
 	}
 }
 
@@ -344,7 +349,7 @@ func (r *Registry) createJobsInBatchWithTxn(
 		start = txn.ReadTimestamp().GoTime()
 	}
 	modifiedMicros := timeutil.ToUnixMicros(start)
-	stmt, args, jobIDs, err := r.batchJobInsertStmt(s.ID(), records, modifiedMicros)
+	stmt, args, jobIDs, err := r.batchJobInsertStmt(ctx, s.ID(), records, modifiedMicros)
 	if err != nil {
 		return nil, err
 	}
@@ -359,7 +364,7 @@ func (r *Registry) createJobsInBatchWithTxn(
 // batchJobInsertStmt creates an INSERT statement and its corresponding arguments
 // for batched jobs creation.
 func (r *Registry) batchJobInsertStmt(
-	sessionID sqlliveness.SessionID, records []*Record, modifiedMicros int64,
+	ctx context.Context, sessionID sqlliveness.SessionID, records []*Record, modifiedMicros int64,
 ) (string, []interface{}, []jobspb.JobID, error) {
 	instanceID := r.ID()
 	const numColumns = 6
@@ -377,7 +382,7 @@ func (r *Registry) batchJobInsertStmt(
 		`claim_session_id`:  func(rec *Record) interface{} { return sessionID.UnsafeBytes() },
 		`claim_instance_id`: func(rec *Record) interface{} { return instanceID },
 		`payload`: func(rec *Record) interface{} {
-			payload := r.makePayload(rec)
+			payload := r.makePayload(ctx, rec)
 			return marshalPanic(&payload)
 		},
 		`progress`: func(rec *Record) interface{} {
@@ -439,7 +444,7 @@ func (r *Registry) CreateJobWithTxn(
 	// TODO(sajjad): Clean up the interface - remove jobID from the params as
 	// Record now has JobID field.
 	record.JobID = jobID
-	j := r.newJob(record)
+	j := r.newJob(ctx, record)
 
 	s, err := r.sqlInstance.Session(ctx)
 	if err != nil {
@@ -476,7 +481,7 @@ func (r *Registry) CreateAdoptableJobWithTxn(
 	// TODO(sajjad): Clean up the interface - remove jobID from the params as
 	// Record now has JobID field.
 	record.JobID = jobID
-	j := r.newJob(record)
+	j := r.newJob(ctx, record)
 	if err := j.runInTxn(ctx, txn, func(ctx context.Context, txn *kv.Txn) error {
 		// Note: although the following uses ReadTimestamp and
 		// ReadTimestamp can diverge from the value of now() throughout a

--- a/pkg/server/server_sql.go
+++ b/pkg/server/server_sql.go
@@ -427,6 +427,7 @@ func newSQLServer(ctx context.Context, cfg sqlServerArgs) (*SQLServer, error) {
 			cfg.clock,
 			cfg.db,
 			cfg.circularInternalExecutor,
+			cfg.ClusterIDContainer,
 			cfg.nodeIDContainer,
 			cfg.sqlLivenessProvider,
 			cfg.Settings,

--- a/pkg/spanconfig/spanconfigjob/job.go
+++ b/pkg/spanconfig/spanconfigjob/job.go
@@ -54,6 +54,18 @@ func (r *resumer) Resume(ctx context.Context, execCtxI interface{}) (jobErr erro
 	}()
 
 	execCtx := execCtxI.(sql.JobExecContext)
+	if !execCtx.ExecCfg().ClusterID().Equal(r.job.Payload().CreationClusterID) {
+		// When restoring a cluster, we don't want to end up with two instances of
+		// the singleton reconciliation job.
+		//
+		// TODO(irfansharif): We could instead give the reconciliation job a fixed
+		// ID; comparing cluster IDs during restore doesn't help when we're
+		// restoring into the same cluster the backup was run from.
+		log.Infof(ctx, "duplicate restored job (source-cluster-id=%s, dest-cluster-id=%s); exiting",
+			r.job.Payload().CreationClusterID, execCtx.ExecCfg().ClusterID())
+		return nil
+	}
+
 	rc := execCtx.SpanConfigReconciler()
 	stopper := execCtx.ExecCfg().DistSQLSrv.Stopper
 

--- a/pkg/sql/gcjob/refresh_statuses.go
+++ b/pkg/sql/gcjob/refresh_statuses.go
@@ -314,12 +314,6 @@ func isProtected(
 			return true, nil
 		}
 
-		// Skip checking the new protected timestamp subsystem if the testing knob
-		// says as such.
-		if execCfg.GCJobTestingKnobs.DisableNewProtectedTimestampSubsystemCheck {
-			return false, nil
-		}
-
 		spanConfigRecords, err := kvAccessor.GetSpanConfigRecords(ctx, spanconfig.Targets{
 			spanconfig.MakeTargetFromSpan(sp),
 		})

--- a/pkg/sql/schema_changer.go
+++ b/pkg/sql/schema_changer.go
@@ -2269,14 +2269,6 @@ type GCJobTestingKnobs struct {
 	// protected timestamp status of a table or an index. The protection status is
 	// passed in along with the jobID.
 	RunAfterIsProtectedCheck func(jobID jobspb.JobID, isProtected bool)
-	// DisableNewProtectedTimestampSubsystemCheck disables checking the new
-	// protected timestamp subsystem when checking the protection status of a
-	// table or an index. This is useful for tests that disable the span
-	// configuration infrastructure, as the new protected timestamp subsystem is
-	// built on top of it.
-	// TODO(arul): Once we've fully migrated all tests to use span configurations
-	// we should be able to get rid of this testing knob as well.
-	DisableNewProtectedTimestampSubsystemCheck bool
 }
 
 // ModuleTestingKnobs is part of the base.ModuleTestingKnobs interface.

